### ISSUE TITLE
[feature request] Create 'fake' hostPort(s)

### ISSUE THREAD

**Issue by mtougeron** [loft-sh/vcluster#250]
It would be nice to have a way for a user of vCluster to create a deployment using hostPort(s) (which is normally for us blocked by a PodSecurityPolicy) and have it create a 'fake' hostPort mapping to the underlying Kubernetes system. This would allow users to test services that would normally use hostPort but not actually expose a hostPort during the testing/validation process.

**Comment by FabianKramm** [comment]
@mtougeron thanks for creating this issue! Interesting idea, how would this fake hostPort mapping look like? What should vcluster respond to connections on these ports?

**Comment by mtougeron** [comment]
The best idea I can come up with at the moment would be a ClusterIP Service on the host k8s system per pod in the vCluster deployment. That way each pod would still have a ip on the service network layer that routes directly to it. I don't know if I'd be entirely happy with that approach but it would kind of simulate what would happen. 

If it helps, I'm digging more into the way vCluster code works this week and might have a better answer once I understand more about how the virtual nodes are handled.

**Comment by olljanat** [comment]
@mtougeron if I understood correctly from https://github.com/loft-sh/vcluster/issues/251#issuecomment-993906742 your use case is that you want to be able to test that manifest which modify it?

One way to do it would be to run policy engine like Kyverno inside of vcluster and mutate those services on fly. Here is example about policy like that:
```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: use-clusterip
spec:
  validationFailureAction: enforce
  background: false
  rules:
  - name: nodeport
    match:
      resources:
        kinds:
        - Service
        namespaces:
        - "vcluster-*"
    preconditions:
      all:
      - key: "{{request.operation}}"
        operator: In
        value:
        - CREATE
      - key: "{{request.object.spec.type}}"
        operator: Equals
        value: NodePort
    mutate:
      patchesJson6902: |-
        - path: "/spec/type"
          op: replace
          value: ClusterIP
        - path: "/spec/externalTrafficPolicy"
          op: remove
```

**Comment by mtougeron** [comment]
Interesting. I'll have to give that a try and see what it can do for the hostPort & hostNetworking flags. Thanks for the suggestion!

**Comment by sgreene570** [comment]
Friendly bump on this feature; it would be really neat and make moving workloads to vcluster easier!
